### ISSUE TITLE
Setting slots before conversation starts leads to empty bot responses

### ISSUE THREAD

**Issue by ricwo** [RasaHQ/rasa#6721]
**Rasa version**: 2.0.0rc1

**Python version**: 3.7.8

**Operating system** (windows, osx, ...): osx

**Issue**: Settings lots via the API for a new conversation ID leads to an empty bot response after the first regular user message. The issue was first reported here: https://forum.rasa.com/t/api-unexpected-behavior-bug/33400

**Comment by chaneyjd** [comment]
I think the tracker is under the mistaken assumption that because it is not empty (but with only SlotSet events) it has interacted with the user (incorrect assumption).

I am assuming also that there might be other events (API) that should be ignored here (or at least not assumed to be user interaction events).

**I am able to "correct" the issue by the following:**
In the applied_events method in trackers.py (core):

**FIND:
---
            else:
                applied_events.append(event)
---
**CHANGE TO:
---
                if not isinstance(event, (SlotSet)):
                    applied_events.append(event)
                else:
                    logger.info("Skipped putting SetSlot Event on applied events queue")
---

**Comment by chaneyjd** [comment]
and the diff:

--- a/rasa/shared/core/trackers.py
+++ b/rasa/shared/core/trackers.py
@@ -425,7 +425,10 @@ class DialogueStateTracker:
                     event.action_name, applied_events
                 )
             else:
-                applied_events.append(event)
+                if not isinstance(event, (SlotSet)):
+                    applied_events.append(event)
+                else:
+                    logger.info("Skipped putting SetSlot Event on applied events queue")

**Comment by ricwo** [comment]
thanks @chaneyjd for looking into this. do you want to open a PR and we can discuss your solution there?

**Comment by chaneyjd** [comment]
sure, let me get a PR today and I'll get with you there

**Comment by chaneyjd** [comment]
issue #6721 fix #6785 PR created ...

**Comment by coderunite** [comment]
This issue still exists in 2.0.3.

```
2020-11-06 06:15:24 INFO     root  - Rasa server is up and running.
2020-11-06 06:16:11 DEBUG    rasa.core.lock_store  - Issuing ticket for conversation 'testbug1'.
2020-11-06 06:16:11 DEBUG    rasa.core.lock_store  - Acquiring lock for conversation 'testbug1'.
2020-11-06 06:16:11 DEBUG    rasa.core.lock_store  - Acquired lock for conversation 'testbug1'.
2020-11-06 06:16:11 DEBUG    rasa.core.tracker_store  - Creating a new tracker for id 'testbug1'.
2020-11-06 06:16:11 DEBUG    rasa.core.lock_store  - Deleted lock for conversation 'testbug1'.
2020-11-06 06:16:49 DEBUG    rasa.core.lock_store  - Issuing ticket for conversation 'testbug1'.
2020-11-06 06:16:49 DEBUG    rasa.core.lock_store  - Acquiring lock for conversation 'testbug1'.
2020-11-06 06:16:49 DEBUG    rasa.core.lock_store  - Acquired lock for conversation 'testbug1'.
2020-11-06 06:16:49 DEBUG    rasa.core.tracker_store  - Recreating tracker for id 'testbug1'
2020-11-06 06:16:50 DEBUG    rasa.core.processor  - Received user message 'hello!' with intent '{'id': 5223738140840328188, 'name': 'greet', 'confidence': 0.9865612983703613}' and entities '[]'
2020-11-06 06:16:50 DEBUG    rasa.core.processor  - Logged UserUtterance - tracker now has 2 events.
2020-11-06 06:16:50 DEBUG    rasa.core.policies.memoization  - Current tracker state [{'user': {'intent': 'greet'}, 'slots': {'user_id': (1.0,)}}]
2020-11-06 06:16:50 DEBUG    rasa.core.policies.memoization  - There is no memorised next action
2020-11-06 06:16:50 DEBUG    rasa.core.policies.mapping_policy  - There is no mapped action for the predicted intent, 'greet'.
2020-11-06 06:16:50 DEBUG    rasa.core.policies.form_policy  - There is no active form
2020-11-06 06:16:50 DEBUG    rasa.core.policies.fallback  - NLU confidence threshold met, confidence of fallback action set to core threshold (0.8).
2020-11-06 06:16:50 DEBUG    rasa.core.policies.ensemble  - Predicted next action using policy_1_TEDPolicy
2020-11-06 06:16:50 DEBUG    rasa.core.processor  - Predicted next action 'action_listen' with confidence 1.00.
2020-11-06 06:16:50 DEBUG    rasa.core.processor  - Action 'action_listen' ended with events '[]'.
2020-11-06 06:16:50 DEBUG    rasa.core.lock_store  - Deleted lock for conversation 'testbug1'.
```

First request:
http://localhost:5005/conversations/testbug1/tracker/events
`{"event":"slot","name":"user_id","value":"test2","timestamp":1604664506.92012}`
```
{
  "sender_id": "testbug1",
  "slots": {
    "communication_recipient": null,
    "communication_type": null,
    "dispute_reason": null,
    "location": null,
    "redirect_type": null,
    "requested_slot": null,
    "user_id": "test2"
  },
  "latest_message": {
    "intent": {},
    "entities": [],
    "text": null,
    "message_id": null,
    "metadata": {}
  },
  "latest_event_time": 1604664506.92012,
  "followup_action": "action_listen",
  "paused": false,
  "events": [
    {
      "event": "slot",
      "timestamp": 1604664506.92012,
      "name": "user_id",
      "value": "test2"
    }
  ],
  "latest_input_channel": null,
  "active_loop": {},
  "latest_action": {},
  "latest_action_name": null
}
```

Second request:
http://localhost:5005/webhooks/rest/webhook
```
{
	"sender":"testbug1",
	"message":"hello!"
}
```

Only response received: `[]`

**Comment by ricwo** [comment]
@coderunite the fix has not been released yet - it will be part of version 2.1

**Comment by coderunite** [comment]
I see 2.1 was released and I don't see a mention of 7020. I also just tested the issue and it still exists.
